android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java

```diff
@@ -455,28 +493,72 @@ public void rename(PluginCall call) {
     }
```

Comment:
this is preventing from copying/renaming if the path is the same but the directory is different, you should compare `fromObject` and  `toObject` instead.

Reply:
Fixed